tests/v2/validation/snapshot/snapshot.go

```diff
@@ -413,3 +414,35 @@ func createPostBackupWorkloads(t *testing.T, client *rancher.Client, clusterID s
 	require.NoError(t, err)
 	require.Equal(t, WorkloadNamePostBackup, postDeploymentResp.ObjectMeta.Name)
 }
+
+// This function waits for retentionlimit+1 automatic snapshots to be taken before verifying that the retention limit is respected
+func createSnapshotsUntilRetentionLimit(t *testing.T, client *rancher.Client, clusterName string, retentionLimit int, timeBetweenSnapshots int) {
+	v1ClusterID, err := clusters.GetV1ProvisioningClusterByName(client, clusterName)
+	if v1ClusterID == "" {
+		v3ClusterID, err := clusters.GetClusterIDByName(client, clusterName)
+		require.NoError(t, err)
+		v1ClusterID = "fleet-default/" + v3ClusterID
+	}
+	require.NoError(t, err)
+
+	fleetCluster, err := client.Steve.SteveType(extdefault.FleetClusterSteveResourceType).ByID(v1ClusterID)
+	require.NoError(t, err)
+
+	provider := fleetCluster.ObjectMeta.Labels["provider.cattle.io"]
+	if provider == "rke" {
```

Comment:
`rke` can be put into a const block as well.

Reply:
This will be handled in a the follow PR to https://github.com/rancher/shepherd/pull/107